src/client/datascience/jupyter/kernels/jupyterKernelService.ts

```diff
@@ -211,9 +219,10 @@ export class JupyterKernelService {
         const specedKernel = kernel as JupyterKernelSpec;
         if (specFile && kernelSpecRootPath) {
             // Spec file may not be the same as the original spec file path.
-            const kernelSpecFilePath = specFile.includes(kernel.name)
-                ? specFile
-                : path.join(kernelSpecRootPath, kernel.name, 'kernel.json');
+            const kernelSpecFilePath =
+                path.basename(specFile).toLowerCase() === kernel.name.toLowerCase()
```

Comment:
This is more accurate
Kernelspec name is the folder that contains the kernelspec file.